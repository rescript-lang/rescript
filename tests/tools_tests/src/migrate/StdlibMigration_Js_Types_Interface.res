/* Implementation to satisfy interface build for tests */

external nullT: Js.null<int> = "nullT"
external nullableT: Js.nullable<string> = "nullableT"
external nullUndefT: Js.null_undefined<float> = "nullUndefT"

external symbolT: Js.Types.symbol = "symbolT"
external objValT: Js.Types.obj_val = "objValT"
external functionValT: Js.Types.function_val = "functionValT"
