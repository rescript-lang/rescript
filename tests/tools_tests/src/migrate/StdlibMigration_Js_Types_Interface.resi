/* Migration tests for Js.res type deprecations */

// Type alias migrations exercised via externals
external nullT: Js.null<int> = "nullT"
external nullableT: Js.nullable<string> = "nullableT"
external nullUndefT: Js.null_undefined<float> = "nullUndefT"

// Js.Types migrations
external symbolT: Js.Types.symbol = "symbolT"
external objValT: Js.Types.obj_val = "objValT"
external functionValT: Js.Types.function_val = "functionValT"
