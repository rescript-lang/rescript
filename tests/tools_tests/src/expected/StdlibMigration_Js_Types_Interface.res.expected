/* Implementation to satisfy interface build for tests */

external nullT: Null.t<int> = "nullT"
external nullableT: Nullable.t<string> = "nullableT"
external nullUndefT: Nullable.t<float> = "nullUndefT"

external symbolT: Symbol.t = "symbolT"
external objValT: Type.Classify.object = "objValT"
external functionValT: Type.Classify.function = "functionValT"

